### PR TITLE
Added REDIS cache configuration property ("username") in cache config…

### PR DESCRIPTION
**Pull request by arfeen**
A new cache configuration property ("username") in cache config for REDIS. Optional field, not needed for REDIS versions do not accept username.

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
@marcovtwout Please take a look when you have time

File: framework/caching/CRedisCache.php [hunk 1]
```diff
@@ -96,8 +101,13 @@ protected function connect()
 		{
 			if($this->ssl)
 				stream_socket_enable_crypto($this->_socket,true,STREAM_CRYPTO_METHOD_TLS_CLIENT);
-			if($this->password!==null)
-				$this->executeCommand('AUTH',array($this->password));
+			if($this->password!==null || $this->username!==null){
+				if(isset($this->username))
```
Comment (rob006) [inline framework/caching/CRedisCache.php:105]:
```suggestion
			if($this->password!==null){
				if($this->username!==null)
```

File: framework/caching/CRedisCache.php [hunk 2]
```diff
@@ -52,6 +53,10 @@ class CRedisCache extends CCache
 	 * @var int the port to use for connecting to the redis server. Default port is 6379.
 	 */
 	public $port=6379;
+		/**
+	 * @var string the username to use to authenticate with the redis server. If set, AUTH command will be sent with username.
+	 */
```
Comment (samdark) [inline framework/caching/CRedisCache.php:58]:
```suggestion
	/**
	 * @var string the username to use to authenticate with the redis server. If set, AUTH command will be sent with username.
	 */
```

File: framework/caching/CRedisCache.php [hunk 3]
```diff
@@ -96,8 +101,13 @@ protected function connect()
...
 			if($this->ssl)
 				stream_socket_enable_crypto($this->_socket,true,STREAM_CRYPTO_METHOD_TLS_CLIENT);
-			if($this->password!==null)
-				$this->executeCommand('AUTH',array($this->password));
+			if($this->password!==null){
+				if($this->username!==null)
+					$this->executeCommand('AUTH',array($this->username, $this->password));
+				else
+					$this->executeCommand('AUTH',array($this->password));
+			}
```
Comment (samdark) [inline framework/caching/CRedisCache.php:109]:
```suggestion
			if ($this->password !== null) {
				if ($this->username !== null) {
					$this->executeCommand('AUTH',array($this->username, $this->password));
				} else {
					$this->executeCommand('AUTH',array($this->password));
				}
			}
```

File: framework/caching/CRedisCache.php [hunk 4]
```diff
@@ -31,6 +31,7 @@
  *             'port'=>6379,
  *             'database'=>0,
  *             'options'=>STREAM_CLIENT_CONNECT,
+ * 			   'username' => 'default' // only for REDIS version 6.0 or later
```
Comment (marcovtwout) [inline framework/caching/CRedisCache.php:34]: This line should use spaces instead of tabs.

**Review by marcovtwout — APPROVED** [review]

**Comment by marcovtwout** [comment]
Thanks!

**Pull request merged**